third_party/libunwindstack/PeCoffUnwindInfos.h

```diff
@@ -68,21 +68,27 @@ struct UnwindInfo {
 
 class PeCoffUnwindInfos {
  public:
-  explicit PeCoffUnwindInfos(PeCoffMemory* pe_coff_memory) : pe_coff_memory_(pe_coff_memory) {}
+  virtual ~PeCoffUnwindInfos() {}
 
   // The value 'unwind_info_file_offset' is the file offset derived from the unwind info
   // offset in the RUNTIME_FUNCTION struct, which is a relative virtual address (RVA). Computing the
   // file offset from the unwind info RVA requires knowledge about the sections of the file, which
   // the class using 'PeCoffUnwindInfos' must use to pass in the right value here.
-  bool GetUnwindInfo(uint64_t unwind_info_file_offset, UnwindInfo* unwind_info);
+  virtual bool GetUnwindInfo(uint64_t unwind_info_file_offset, UnwindInfo* unwind_info) = 0;
   ErrorData GetLastError() const { return last_error_; }
 
- private:
-  bool ParseUnwindInfoAtOffset(uint64_t file_offset, UnwindInfo* unwind_info);
-
-  PeCoffMemory* pe_coff_memory_;
+ protected:
   ErrorData last_error_{ERROR_NONE, 0};
+};
 
+class PeCoffUnwindInfosImpl : public PeCoffUnwindInfos {
+ public:
+  explicit PeCoffUnwindInfosImpl(Memory* memory) : pe_coff_memory_(memory) {}
+  bool GetUnwindInfo(uint64_t unwind_info_file_offset, UnwindInfo* unwind_info) override;
+
+ private:
+  bool ParseUnwindInfoAtOffset(uint64_t file_offset, UnwindInfo* unwind_info);
+  PeCoffMemory pe_coff_memory_;
   // This is a cache of unwind infos.
   std::unordered_map<uint64_t, UnwindInfo> unwind_info_offset_to_unwind_info_;
 };
```

Comment:
It's not really an implementation class if it is defined in a public header. Would exposing a factory function suffice in this case or do you really need the `Impl` class here?

Reply:
(This comment also applies to the other Impl classes. As far as I've seen they are also declared in a public header.)

Reply:
Yes, you are right, will add a factory function to not expose the Impl classes in the header.

Reply:
Ok, should be addressed now. Let me know if this looks alright.